### ISSUE TITLE
Incorrect HTTP Status detection

### ISSUE THREAD

**Issue by sergioredondo** [nazuke/SEOMacroscope#15]
I've crawled my website with Macroscope and all my html pages are detected with 301 as status code, when all these pages actually return a 200 response.

**Comment by jd440** [comment]
same for me, I cheched with screaming, xenu, and curl -I
and seomacroscope return me 301 and all other tools return 200,

And something even more strange seomacroscope return me a 301 from and to the same url
http://example.com/a -301-> http://example.com/a

**Comment by sergioredondo** [comment]
Exactly!

**Comment by jd440** [comment]
I downgrade to v1.5 no problem

**Comment by jd440** [comment]
checking log I notice the V1.6 does make GET request only HEAD

```
255.ip-255-255-255.eu - - [25/Jan/2018:08:45:23 +0100] "HEAD /mon-url/ HTTP/1.1" 301 - "-" "SEO Macroscope/1.6.0.0"

255.ip-255-255-255.eu - - [25/Jan/2018:09:26:45 +0100] "GET /mon-url/ HTTP/1.1" 200 38953 "-" "SEO Macroscope/1.5.0.0"
255.ip-255-255-255.eu - - [25/Jan/2018:09:26:46 +0100] "HEAD /mon-url/ HTTP/1.1" 200 - "-" "-"
```

**Comment by nazuke** [comment]
Many thanks for your feedback @sergioredondo and @jd440 

These are very likely to be bugs. However, a a lot has changed since I've been working on the v1.7 branch.

Would I be able to ask what the URLs were that SEO Macroscope was having trouble with please?

And would it be permitted to run one or two tests against those URLs?

**Comment by nazuke** [comment]
Thanks again for your feedback @jd440,

To explain what's going on with the mix of HEAD and GET requests, SEO Macroscope will always issue a HEAD request against each URL in a scan.

This verifies that the URL in question is available, or not. This also gathers information about the URL, such as its MIME Type, internal/external status, whether it's a redirect, etc.

In those cases where the URL is something that has been set to not be processed, such as a large PDF file, the HEAD request still checks its availability, but it will not be downloaded and processed further. In the case of PDFs, "not processing" means that things like the PDF title won't be extractable. It's a trade off. HTML pages will always receive a HEAD, and then a GET request made against them.

Likewise, "external" URLs, such as where you have a link to a third-party site, will only receive a HEAD request. Unless the preferences are set to follow external links.

Actually, this is why it's always a good idea to only use one or two threads at first, to save on load on your web server.

**Comment by sergioredondo** [comment]
In my case, all the html pages of my personal website sergioredondo.com. You can run a test aginst them without problem.

**Comment by jd440** [comment]
@nazuke 
i don't think this coming from http2 support as my server doesn't support it yet and my website is in http


furtermore I don't have any trouble on v1.5 but on 1.6 all url return 301

As you could see from my log
For the same url
```
255.ip-255-255-255.eu - - [25/Jan/2018:08:45:23 +0100] "HEAD /mon-url/ HTTP/1.1" 301 - "-" "SEO Macroscope/1.6.0.0"

255.ip-255-255-255.eu - - [25/Jan/2018:09:26:45 +0100] "GET /mon-url/ HTTP/1.1" 200 38953 "-" "SEO Macroscope/1.5.0.0"
255.ip-255-255-255.eu - - [25/Jan/2018:09:26:46 +0100] "HEAD /mon-url/ HTTP/1.1" 200 - "-" "-"
```
The server is reply by a 301 to v1.6 and 200 on V1.6

About my website, give me your mail (i don't wich to display it here.)

**Comment by nazuke** [comment]
@jd440 I just realized, those 301 statuses are in your web server log. So I'm not quite sure what might be happening there.

I understand your privacy concerns, all email correspondence for SEO Macroscope may be sent to the address here please:

> seo.macroscope@gmail.com

This is also listed on the site here:

> https://nazuke.github.io/SEOMacroscope/

Many thanks for your help, and for your interest in SEO Macroscope.

**Comment by nazuke** [comment]
@sergioredondo Many thanks for your help with this issue.

I'll investigate what may be happening when SEO Macroscope crawls your site shortly.

Please note, the DEV user-agent will appear as:

> SEO-Macroscope-DEVELOPER-MODE

...in your logs.

I should probably add something in the README about specifying the problematic URL in these issues. It helps a lot. There are an infinite number of websites out there, and the strangest things can happen sometimes. I wrote this for myself originally, to help with my own work; so it's mostly been used against large corporate sites, but it'd be nice if this were to useful for others too.

**Comment by jd440** [comment]
@nazuke Hi just sent you an email

**Comment by jd440** [comment]
@nazuke Did you get my answer?
